src/App.tsx

```diff
@@ -1,29 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = items.length;
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currPage, setCurrPage] = useState(1);
+  const numOfPages = Math.ceil(items.length / itemsPerPage);
+
+  const elements = currPage * itemsPerPage;
+
+  const fromItem = (currPage - 1) * itemsPerPage + 1;
+  const toItem = elements > total ? total : elements;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currPage} (items {fromItem} - {toItem} of {total})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            onChange={event => {
+              setItemsPerPage(+event.target.value);
+              setCurrPage(() => {
+                const defaultPage = 1;
+
+                window.location.hash = `#${defaultPage}`;
+
+                return defaultPage;
+              });
+            }}
+            defaultValue={5}
+          >
+            {[3, 5, 10, 20].map((option, numb) => (
+              <option value={option} key={numb}>
```

Comment:
Move this array to the variable and use it here
```suggestion
            {[3, 5, 10, 20].map((option) => (
              <option value={option} key={option}>
```